src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,135 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+import duke.ui.Ui;
+
+
+/**
+ * Task List to store all the tasks
+ */
+public class TaskList {
+
+    protected ArrayList<Task> taskList;
+
+    public TaskList() {
+        this.taskList = new ArrayList<>();
+    }
+
+    /**
+     * Initialises the task list
+     *
+     * @param file The file storing the task information
+     */
+    public TaskList(File file) {
+        this.taskList = new ArrayList<>();
+        try {
+            Scanner s = new Scanner(file); // create a Scanner using the File as the source
+            while (s.hasNext()) {
+                String cur = s.nextLine();
+
+                String[] line = cur.split("\\|");
+
+                switch (line[0].trim()) {
+                case "T":
+                    taskList.add(new Todo(line[2].trim()));
+                    if (line[1].equals("1")) {
+                        taskList.get(taskList.size() - 1).markAsDone();
+                    }
+                    break;
+                case "D":
+                    LocalDate time = LocalDate.parse(line[3].trim());
+                    taskList.add(new Deadline(line[2].trim() + " ", time));
+                    if (line[1].trim().equals("1")) {
+                        taskList.get(taskList.size() - 1).markAsDone();
+                    }
+                    break;
+                case "E":
+                    taskList.add(new Event(line[2].trim() + " ", line[3].trim()));
+                    if (line[1].trim().equals("1")) {
+                        taskList.get(taskList.size() - 1).markAsDone();
+                    }
+                    break;
+                default:
+                }
+            }
+        } catch (FileNotFoundException e) {
+            Ui.fileExceptionUi();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Deletes a task from the list
+     *
+     * @param index The index of the task in the list
+     * @return The task deleted
+     */
+    public Task delete(int index) {
+        Task removed = taskList.remove(index);
```

Comment:
I feel that it might be better to use the name `removedTask` instead :)
I would sat the same about instances in several other places as well, (e.g. `markedTask`, `unmarkedTask`, etc.)